src/main/java/igrad/model/requirement/Requirement.java

```diff
@@ -92,14 +92,46 @@ public boolean hasModule(Module module) {
         return modules.contains(module);
     }
 
+    /**
+     * Returns true if any modules in {@code modules} with the same identity as {@code module} exists in the list.
+     */
+    public boolean hasModule(List<Module> modules) {
+        requireNonNull(modules);
+
+        return this.modules.contains(modules);
+    }
+
     /**
      * Adds a {@code module} to the list.
      * The module must not already exist in the list.
      */
-    public void addModule(Module module) {
+    public int addModule(Module module) {
```

Comment:
i think it doesnt make sense to return the credits value from adding the module. i mean the person using the function understands. but it doesnt make sense from a design perspective.
like why cant you just directly get the credits from the module? since you have the module to add you can just do a getcredits call and it would be the same. it doesnt make sense to go through requirements to get the credits of the module...

Reply:
yeap, i've changed it back in the latest one

Reply:
cos I was under the impression that we need to validate if requirements is full (i.e, adding too many mods under requirement). but then just now you told me no  need

Reply:
(but imagine instead, that we've to validate whether requirements are 'full', then it makes plausible sense for the `public int addModule(List<Module> modules)` method, to iterate through that `modules` list to compute the total MCs, for us, as it is adding those  modules to the specified requirement)

Now, as for the `public int addModule(Module module)` method, I think i kinda agree with you could actually just get the `Credits` directly from modules itself, instead of that method returning the credits. But, if our`public  int addModule(List<Module> modules)` is already returning an `int`, this would result in an inconsistency with the method that only adds one module.

All this talk about the design issue, is based on the assumption that we need to validate if requirements are 'full', but as you've told  me, we don't have to.

Thus I've reverted it back to void (for both methods). You can check the latest code

Reply:
hmmm to me it still doesnt make sense. if i were to write this, i would get the list outside and iterate through the new list outside as well to get calculate the total credits. cos to me things like requirement and credits are all just a container, and even more so since its just part of the model. shouldnt the logic class be handling the logic, afterall?

Reply:
oh yeah that's true

Reply:
logic should be handling logic, but somehow i feel all these  are  low-level logic which Logic shouldn't be concerned abt hmm

Reply:
thanks btw for taking time to look through  my code and giving me comments